lib/inventory.dart

```diff
@@ -0,0 +1,176 @@
+                                      products(
+                                        categories.data?['data'][index]['id'],
+                                      ),
+                                    ],
+                                  ),
+                                ),
+                              )
+                            ],
+                          );
+                        });
+                      },
+                    ),
+                  );
+              }
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  Widget products(int categoryId) {
+    var temp = api.fetchInventoryByUserLocation(categoryId);
+    return StatefulBuilder(
+      builder: (context, setState) {
+        return FutureBuilder(
+          future: temp,
+          builder: (BuildContext context,
+              AsyncSnapshot<Map<String, dynamic>> products) {
+            switch (products.connectionState) {
+              case ConnectionState.waiting:
+                return const Text('Loading...');
+              default:
+                return SizedBox(
+                  child: Column(
+                    children: <Widget>[
+                      ListView.builder(
+                        shrinkWrap: true,
+                        itemCount: products.data?['data'].length,
+                        itemBuilder: (BuildContext context, int index) {
+                          return Column(
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            mainAxisSize: MainAxisSize.min,
+                            children: <Widget>[
+                              ListTile(
+                                title: Text(
+                                    '${products.data?['data'][index]['name']} ${products.data?['data'][index]['content']} ${products.data?['data'][index]['unit'] ?? ''}'),
+                                trailing: const Icon(Icons.edit),
+                              ),
+                            ],
+                          );
+                        },
+                      ),
+                      Row(
+                        crossAxisAlignment: CrossAxisAlignment.center,
+                        mainAxisSize: MainAxisSize.min,
+                        children: [
+                          Container(
+                            margin: const EdgeInsets.only(bottom: 10, right: 5),
+                            child: ElevatedButton(
+                              onPressed: () {
+                                if (products.data?['links']['prev'] != null) {
+                                  setState(() {
+                                    temp = api.fetchPage(
+                                        products.data?['meta']['path'],
+                                        products.data?['meta']['current_page'] -
+                                            1,
+                                        categoryId);
+                                  });
```

Comment:
If there is no previous and for that matter next, we want the button to be disabled.
To do this simply make the onpressed be null.
eg:
```
if (state != null) {
    Do stuff
} else {
    null;
}
```